test/metadata_test.go

```diff
@@ -77,13 +103,19 @@ func (s) TestInvalidMetadata(t *testing.T) {
 			}
 			test := tests[testNum]
 			testNum++
-			if err := stream.SetHeader(test.md); !reflect.DeepEqual(test.want, err) {
-				return fmt.Errorf("call stream.SendHeader(md) validate metadata which is %v got err :%v, want err :%v", test.md, err, test.want)
+			// merge original md and added md
+			md := test.md.Copy()
+			for i := 0; i < len(test.appendMD); i += 2 {
+				md.Append(test.appendMD[i], test.appendMD[i+1])
 			}
```

Comment:
Ah, I was mixing up `Append` and `Join`, sorry.  So this could be simplified to:

```go
// Merge original md and added md.
md := metadata.Join(test.md, metadata.Pairs(test.appendMD...))
```